glean-core/rlb/src/private/memory_distribution.rs

```diff
@@ -0,0 +1,147 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use inherent::inherent;
+use std::sync::Arc;
+
+use glean_core::metrics::{DistributionData, MemoryUnit, MetricType};
+use glean_core::ErrorType;
+
+use crate::dispatcher;
+
+// We need to wrap the glean-core type: otherwise if we try to implement
+// the trait for the metric in `glean_core::metrics` we hit error[E0117]:
+// only traits defined in the current crate can be implemented for arbitrary
+// types.
+
+/// This implements the developer-facing API for recording memory distribution metrics.
+///
+/// Instances of this class type are automatically generated by the parsers
+/// at build time, allowing developers to record values that were previously
+/// registered in the metrics.yaml file.
+#[derive(Clone)]
+pub struct MemoryDistributionMetric(pub(crate) Arc<glean_core::metrics::MemoryDistributionMetric>);
+
+impl MemoryDistributionMetric {
+    /// The public constructor used by automatically generated metrics.
+    pub fn new(meta: glean_core::CommonMetricData, memory_unit: MemoryUnit) -> Self {
+        Self(Arc::new(
+            glean_core::metrics::MemoryDistributionMetric::new(meta, memory_unit),
+        ))
+    }
+}
+
+#[inherent(pub)]
+impl glean_core::traits::MemoryDistribution for MemoryDistributionMetric {
+    /// Accumulates the provided sample in the metric.
+    ///
+    /// # Arguments
+    ///
+    /// * `sample` - The sample to be recorded by the metric. The sample is assumed to be in the
+    ///   configured memory unit of the metric.
+    ///
+    /// ## Notes
+    ///
+    /// Values bigger than 1 Terabyte (2<sup>40</sup> bytes) are truncated
+    /// and an `ErrorType::InvalidValue` error is recorded.
```

Comment:
Because @badboy found out that rustdocs correctly refer us to the trait docs if docstrings are missing here, he did not add docstrings to [events](https://searchfox.org/glean/source/glean-core/rlb/src/private/event.rs#47). Maybe we should to the same here? WDYT @badboy ? (and to other comments in this trait implementation.

Reply:
If so, that'd help  me out because I have a broken link in the rustdocs as they're written : |

Reply:
Yup, rustdocs will give a link to the trait where the docs are then visible, that's why I left them out on events. I didn't go around removing them anywhere else, thought that could be done afterwards.
For new stuff IMO it's fine to leave them out

Reply:
I filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1681570) to do the follow-up work on removing docstrings where possible.